OpenRA.Mods.Common/Traits/Attack/AttackDetonates.cs

```diff
@@ -11,33 +11,33 @@
 using System.Collections.Generic;
 using System.Drawing;
 using OpenRA.Activities;
-using OpenRA.Mods.Common;
 using OpenRA.Mods.Common.Activities;
 using OpenRA.Mods.Common.Orders;
-using OpenRA.Mods.Common.Traits;
 using OpenRA.Traits;
 
-namespace OpenRA.Mods.RA.Traits
+namespace OpenRA.Mods.Common.Traits
 {
-	class DemoTruckInfo : ITraitInfo, Requires<ExplodesInfo>
+	[Desc("Allows a detonate attack where it moves next to the target and then explodes.")]
+	class AttackDetonatesInfo : ITraitInfo, Requires<ExplodesInfo>
 	{
 		[VoiceReference] public readonly string Voice = "Action";
 
-		public object Create(ActorInitializer init) { return new DemoTruck(init.Self, this); }
+		public object Create(ActorInitializer init) { return new AttackDetonates(init.Self, this); }
 	}
 
-	class DemoTruck : IIssueOrder, IResolveOrder, IOrderVoice
+	class AttackDetonates : IIssueOrder, INotifyCreated, IResolveOrder, IOrderVoice
```

Comment:
I assume this was required so that a lint test will automatically spot a regression when `Explodes` is accidentally removed although indeed it limits flexibility.